src/Ion.ts

```diff
@@ -57,7 +57,7 @@ function makeBinaryReader(span: Span, options: Options) : BinaryReader {
   return new BinaryReader(span, options && options.catalog);
 }
 
-function makeTextReader(span: Span, options: Options) : TextReader {
+function makeTextReader(span: StringSpan, options: Options) : TextReader {
```

Comment:
yes this is simply to allow high/low surrogate pairing without being tied to the state of the span.
Simply put the span needs to be rebuilt from the ground up and stringspan/binaryspan should have no relation.
Eventually this will involve reimplementing stan's work on tokenization and removal of the span altogether.